operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java

```diff
@@ -763,23 +766,83 @@ private boolean haveServerData() {
               StatusUpdateContext.this::isNonClusteredServer).collect(Collectors.toList());
         }
 
-        private boolean allNonClusteredServersRunning() {
-          return getNonClusteredServers().stream().allMatch(StatusUpdateContext.this::isServerReady);
-        }
-
         private boolean allIntendedServersReady() {
           return haveServerData()
               && allStartedServersAreComplete()
               && allNonStartedServersAreShutdown()
               && serversMarkedForRoll().isEmpty();
         }
 
-        private boolean sufficientServersRunning() {
-          return atLeastOneApplicationServerStarted() && allNonClusteredServersRunning() && allClustersAvailable();
+        private DomainCondition createAvailableCondition() {
+          final List<String> unreadyNonClusteredServers = getUnreadyNonClusteredServers();
+          final List<ClusterCheck> unavailableClusters = getUnavailableClusters();
+
+          if (noApplicationServersReady()) {
+            return createNotAvailableCondition(LOGGER.formatMessage(NO_APPLICATION_SERVERS_READY));
+          } else if (!unreadyNonClusteredServers.isEmpty()) {
+            return createNotAvailableCondition(
+                LOGGER.formatMessage(NON_CLUSTERED_SERVERS_NOT_READY, formatServers(unreadyNonClusteredServers)));
+          } else if (!unavailableClusters.isEmpty()) {
```

Comment:
Not a show stopper, but it's possible to have _both_ unready non-clustered servers and unavailableClusters at the same time, but, per the above logic, only the former will be included in the message when this happens.

Reply:
Yes, correct. The idea is not to give every possible bit of information. That can actually be gleaned from the status as a whole. Rather, the idea is to give users some quick guidance of what they should be looking for.

Reply:
Have you tried the branch to see if it gives you the messages needed to diagnose the intermittent problems you were seeing?

Reply:
Can you save me a little time and tell me which particular one?  (There are more than one.)